### PR TITLE
Source NYT

### PR DESCRIPTION
**Pull request by ibhuiyan17**
PR for adding the New York Times timeseries as a source, mentioned in #224 

todo
- [x] Add unit tests for NYT source
- [x] Working v2 endpoints
- [x] Generate mock data
- [x] Add regression tests
- [x] Update documentation

** removed "Working v1 endpoints" from todo since current implementation of v1 doesn't support different sources

### CODE REVIEW

**Comment by ibhuiyan17** [comment]
Yep, still a work in progress, but thanks for pointing this out!

**Comment by nischalshankar** [comment]
There were a few things we wanted to check in with you guys.

Output for NYT:
Currently, the NYT source doesn't provide coordinates, so we have just added null to those output values. NYT provides a fips number (specifies geographic location), which we could add to the output directly or potentially use it to translate to coordinates. Another option is we could use the (county, state) coordinates from csbs to find the coordinates of NYT locations, but this could be an inefficient solution. A way we could do this might be to store the specific coordinates in a file or a database.

Testing:
We tested all the v2 endpoints for NYT. The only content check we added is done for get locations which follows the same format as the other sources. Let us know if you want any additional content checks for any of the other sources. Also, we did not check for the specific value of country population since it could possibly change.

File: app/services/location/nyt.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    data = list(csv.DictReader(text.splitlines()))
+
+    # Group together locations (NYT data ordered by dates not location).
+    grouped_locations = get_grouped_locations_dict(data)
+
+    # The normalized locations.
+    locations = []
+
+    idx = 0
+    for county_state, histories in grouped_locations.items():
```
Comment (Kilo59) [inline app/services/location/nyt.py:89]: Should use `enumerate()` for this.
```python
 for idx, (county_state, histories) in enumerate(grouped_locations.items()):
```
https://dbader.org/blog/python-enumerate

File: README.md [hunk 2]
```diff
@@ -24,18 +24,24 @@ Support multiple data-sources.
 ![Covid-19 Recovered](https://covid19-badges.herokuapp.com/recovered/latest)
 ![Covid-19 Deaths](https://covid19-badges.herokuapp.com/deaths/latest)
 
+## New York Times is now available as a source!
+
+**Specify source parameter with ?source=nyt. NYT also provides a timeseries! To view timelines of cases by US counties use ?source=nyt&timelines=1**
```
Comment (Kilo59) [inline README.md:29]: I would use a boolean for the example. Even though `1` will work, I think an explicit `true` is more clear.
`source=nyt&timelines=true`

**Comment by ibhuiyan17** [comment]
@Kilo59 

Just finished resolving the points you commented on. Please let me know if there's anything else, thanks!

**Comment by Kilo59** [comment]
@all-contributors please add @nischalshankar for code, docs

**Comment by Kilo59** [comment]
@all-contributors please add @ibhuiyan17 for docs

**Pull request merged**